fatiando/mesher.py

```diff
@@ -1721,3 +1721,594 @@ def keep(cell):
+        '''
+        Calculate the susceptibility tensor (in SI) in the main system.
+
+        The susceptibility tensor is calculated if 'susceptibility tensor'
+        is defined in the dictionary of physical properties props.
+        In this case, 'susceptibility tensor' must have six elements:
+        three positive eigenvalues (principal susceptibilities), in
+        descending order, and three angles (in degrees) defining
+        the eigenvector matrix U of the susceptibility tensor, respectively.
+        The eigenvector matrix is defined by the function
+        coord_transf_matrix_oblate.
+        '''
+
+        assert 'susceptibility tensor' in self.props, 'susceptibility tensor \
+is not in the dictionary of physical properties'
+
+        assert len(self.props['susceptibility tensor']) == 6, 'susceptibili\
+ty tensor must be a list containing six elements'
+
+        # Large, intermediate and small eigenvalues of the
+        # susceptibility tensor (principal susceptibilities)
+        k1 = self.props['susceptibility tensor'][0]
+        k2 = self.props['susceptibility tensor'][1]
+        k3 = self.props['susceptibility tensor'][2]
+
+        assert k1 >= k2 >= k3, 'the eigenvalues must be given in \
+descending order'
+
+        assert (k1 > 0) and (k2 > 0) and (k3 > 0), 'the eigenvalues must \
+be all positive'
+
+        # Angles (in degrees) defining the eigenvector matrix
+        # of the susceptibility tensor
+        strike = self.props['susceptibility tensor'][3]
+        dip = self.props['susceptibility tensor'][4]
+        rake = self.props['susceptibility tensor'][5]
+
+        # Eigenvector matrix of the susceptibility tensor
+        alpha, gamma, delta = auxiliary_angles(strike, dip, rake)
+        U = coord_transf_matrix_oblate(alpha, gamma, delta)
+
+        suscep_tensor = numpy.dot(U, numpy.diag([k1, k2, k3]))
+        suscep_tensor = numpy.dot(suscep_tensor, U.T)
+
+        return suscep_tensor
+
+
+def auxiliary_angles(strike, dip, rake):
+    '''
+    Calculate auxiliary angles alpha, gamma and delta (Clark et al., 1986)
+    as functions of geological angles strike, dip and rake
+    (Clark et al., 1986; Allmendinger et al., 2012), given in degrees.
+    This function implements the formulas presented by
+    Clark et al. (1986).
+
+    References:
+
+    Clark, D., Saul, S., and Emerson, D.: Magnetic and gravity anomalies
+    of a triaxial ellipsoid, Exploration Geophysics, 17, 189-200, 1986.
+
+    Allmendinger, R., Cardozo, N., and Fisher, D. M.:
+    Structural geology algorithms : vectors and tensors,
+    Cambridge University Press, 2012.
+    '''
+
+    strike_r = numpy.deg2rad(strike)
+    cos_dip = numpy.cos(numpy.deg2rad(dip))
+    sin_dip = numpy.sin(numpy.deg2rad(dip))
+    cos_rake = numpy.cos(numpy.deg2rad(rake))
+    sin_rake = numpy.sin(numpy.deg2rad(rake))
+
+    aux = sin_dip*sin_rake
+    aux1 = cos_rake/numpy.sqrt(1 - aux*aux)
+    aux2 = sin_dip*cos_rake
+
+    if aux1 > 1.:
```

Comment:
This kind of check won't work if any of the inputs are numpy arrays but we don't need to worry about that now.